log_cache.go

```diff
@@ -33,6 +33,17 @@ func NewLogCache(capacity int, store LogStore) (*LogCache, error) {
 	return c, nil
 }
 
+// IsMonotonic implements the MonotonicLogStore interface. This is a shim to
+// expose the underyling store as monotonically indexed or not.
+func (c *LogCache) IsMonotonic() bool {
+	store, ok := c.store.(MonotonicLogStore)
+	if ok {
```

Comment:
nit: I don't have a strong preference but some Gophers prefer to restrict the scope and do this in one line like:

```suggestion
	if store, ok := c.store.(MonotonicLogStore); ok {
```